src/Terrain/Location.java

```diff
@@ -51,6 +51,16 @@ public boolean isCaptureable()
     return (environs.terrainType == Environment.Terrains.CITY || environs.terrainType == Environment.Terrains.FACTORY || environs.terrainType == Environment.Terrains.AIRPORT || environs.terrainType == Environment.Terrains.SEAPORT || environs.terrainType == Environment.Terrains.HQ || environs.terrainType == Environment.Terrains.LAB);
   }
 
+  /**
+   * @return true if this Location has an industry, false else.
+   */
+  public boolean isIndustry()
+  {
+    return (environs.terrainType == Environment.Terrains.FACTORY ||
```

Comment:
Rather than explicitly checking a set of enum values, is there a more generic/change-proof way to check if the CO can build something from here? Maybe pass the CO as a parameter and then just return `!co.getShoppingList(environs.terrainType).isEmpty()` or something.

Reply:
Yup. We can absolutely do that. Deleting this function and using that instead since if the Location has no CO... then it doesn't know whether it's an industry.
It's true that we only care whether a CO can buy from this territory, but that may not always be the case and doing it outside of this class makes that more explicit IMO.